sql/2020/22_HTTP_2/percentage_of_h2_and_h3_sites_affected_by_cdn_prioritization.sql

```diff
@@ -0,0 +1,36 @@
+# standardSQL
+# Percentage of H2 and H3 sites affected by CDN prioritization issues
+SELECT 
+  client,
+  http_version,
+  IF(pages.cdn = "", "Not using CDN", pages.cdn) AS CDN,
+  IF(prioritization_status IS NOT NULL, prioritization_status, "Unknown") prioritizes_correctly,
+  COUNT(0) AS num_pages,
+  ROUND(COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client), 4) AS pct
+FROM (
+    SELECT
+      date,
+      client,
+      JSON_EXTRACT_SCALAR(payload, '$._protocol') as http_version,
+      url,
+      _cdn_provider as cdn
```

Comment:
One thing about this field is that there may be a chain of CDNs, for example `Cloudflare, Google`. Do you need to parse one of those out? Otherwise it won't be able to be joined with the per-CDN values in `h2_prioritization_cdns`. cc @bazzadp 

Here are the [saved results](https://docs.google.com/spreadsheets/d/1M1tijxf04wSN3KU0ZUunjPYCrVsaJfxzuRCXUeRQ-YU/edit#gid=1152953475) for this query for reference.

Reply:
@rviscomi I understand your point and the short answer is I do not know the answer right now. I will have to speak with either @bazzadp or @paulcalvano to get their thoughts

Reply:
Yeah I didn't fully get what those values were to be honest. Chained CDNs makes sense. The main one will be the one that the user hits - is that the first one in the last or the last one? @gregorywolf could you investigate a few examples?

Also are there many of these? If not maybe just ignore as noise. We never noticed this last year (and btw also missed out on WordPress last year - worth noting in the results @gregorywolf as imagine that alone could account for a measurable uptick).

Reply:
Looks like 0.478% of sites are using chained CDNs. I don't think that's significant enough to worry about to be honest. And chaining will introduce all sorts of other prioritisation weirdness anyway. So I'd ignore.

Reply:
@andydavies or @pmeenan would know more about the CDN value chain and what the positions mean.

See the [results sheet](https://docs.google.com/spreadsheets/d/1M1tijxf04wSN3KU0ZUunjPYCrVsaJfxzuRCXUeRQ-YU/edit#gid=1152953475) for a sense of how common this is. It's in the ~thousands of pages.

Reply:
The position doesn't mean anything other than the order they were detected when looking at the CNAME and headers.

Reply:
OK let's ignore but add a note that chained CDNs will not include correct results in this chapter.

To be honest, both would need to support this to get optimal usage (and even then it's questionable!) and only the "Cloudflare, Fastly" one will from a quick glance meaning 0.13% of sites are affected which is even smaller.

So I say ignore but make a comment on the sheet (maybe linking to this conversation?)